### DISCUSSION TITLE
Create Geoid from Internal Resource in project

### DISCUSSION THREAD

**Discussion by Rangalon** [noelex/GeographicLib.NET#22] [Ideas]
Hello,

I'm using a Geoid from a file stored on my disk (ProgramData). But to share my application with my users, i must copy the file on each user's computers.

The file is included in the project as a resource (Byte[]), but it's not possible to create the geoid from a Byte[]. Could you add the possibility to do so?

If you want, you can grant me the rights to do it myself, it would be a pleasure for me to do so!

Romain

**Comment by ShawnStoddard** [comment]
Do you have an example of what you’re doing?

**Reply by Rangalon** [reply]
I've written a library to handle 3d Data and also WGS84 model for GPS positionning.

I'm trying to add Geoid 2008 information to enhance the quality of information provided by my application for my users.

Then, I'd like to add the pgm file as a resource in the project (it will be a byte[]), and then create the "Geoid" object from the "byte[]" stored in the library.

As is, i must first copy the file on the user's computer, then create the Geoid from the file written on the disk... I would like to left as less stuff as possible on user's computers...

**Comment by noelex** [comment]
Actually you can distribute the geoid file along with your executable, and specify the location of that geoid file via `path` argument when instantiating a Geoid.
If you prefer embedding the geoid file in your application, I can add constructor overloads to Geoid class which will allow you to create `Geoid` instances from byte arrays and streams.

**Reply by Rangalon** [reply]
That's exactly what i would like to have!
